wire-ios/Wire-iOS/Sources/UserInterface/Settings/DeviceManagement/ClientListViewController.swift

```diff
@@ -215,14 +217,18 @@ final class ClientListViewController: UIViewController,
             e2eiCertificateEnrollment: userSession.enrollE2EICertificate
         )
         viewModel.showCertificateUpdateSuccess = {[weak self] certificateChain in
-            self?.updateAllClients()
+            self?.updateAllClients {
+                self?.updateE2EIdentityCertificateInDetailsView()
+            }
+
             let successEnrollmentViewController = SuccessfulCertificateEnrollmentViewController()
             successEnrollmentViewController.certificateDetails = certificateChain
             successEnrollmentViewController.onOkTapped = { viewController in
                 viewController.dismiss(animated: true)
             }
             successEnrollmentViewController.presentTopmost()
         }
+        deviceInfoViewModel = viewModel
```

Comment:
question: would it work to have the viewModel DeviceDetailsView as StateObject instead of ObservableObject?

Reply:
Based on Apple's recommendation, that should be an observable object. I could give it a try and see how it performs. [Documentation](https://developer.apple.com/documentation/swiftui/stateobject)
```
Use a state object as the single source of truth for a reference type that you store in a view hierarchy. Create a state object in an [App](https://developer.apple.com/documentation/swiftui/app), [Scene](https://developer.apple.com/documentation/swiftui/scene), or [View](https://developer.apple.com/documentation/swiftui/view) by applying the @StateObject attribute to a property declaration and providing an initial value that conforms to the [ObservableObject](https://developer.apple.com/documentation/Combine/ObservableObject) protocol. Declare state objects as private to prevent setting them from a memberwise initializer, which can conflict with the storage management that SwiftUI provides
```

Reply:
where is your single source of truth?